src/backend/distributed/shardsplit/shardsplit_decoder.c

```diff
@@ -87,29 +91,30 @@ split_change_cb(LogicalDecodingContext *ctx, ReorderBufferTXN *txn,
 	}
 
 	Oid targetRelationOid = InvalidOid;
+	HeapTuple existingTuple = NULL;
 	switch (change->action)
 	{
 		case REORDER_BUFFER_CHANGE_INSERT:
 		{
-			HeapTuple newTuple = &(change->data.tp.newtuple->tuple);
-			targetRelationOid = FindTargetRelationOid(relation, newTuple,
+			existingTuple = &(change->data.tp.newtuple->tuple);
+			targetRelationOid = FindTargetRelationOid(relation, existingTuple,
 													  replicationSlotName);
 			break;
 		}
 
 		/* updating non-partition column value */
 		case REORDER_BUFFER_CHANGE_UPDATE:
 		{
-			HeapTuple newTuple = &(change->data.tp.newtuple->tuple);
-			targetRelationOid = FindTargetRelationOid(relation, newTuple,
+			existingTuple = &(change->data.tp.newtuple->tuple);
```

Comment:
I think existingTuple is a bit confusing in this case, since I would expect that to be the old one. 

Maybe mainTuple or something? (or not touch this switch and do the GetTupleForTargetSchema for each case separately)

Reply:
@marcocitus: I checked this behavior and below is my understanding from the PG code. There is some different behavior  for oldtuple in case of UPDATE and DELETE.

1) **oldtuple w.r.t UPDATE operation:**
      a) For REPLICA_IDENTITY_FULL, oldtuple is always valid.
      b) However, when its not the case, it may or may not be present. Check this line in [logicalrep_write_update](https://github.com/postgres/postgres/blob/master/src/backend/replication/logical/proto.c#:~:text=if%20(oldslot%20!%3D%20NULL)). Due to this, I was getting a fault initially for replica identity with primary key combination. I was assuming oldtuple in UPDATE would always be present and trying to reformat it according to new schema
      
2) **oldtuple w.r.t DELETE operation:**
The oldtuple is always present for DELETE operation. [logicalrep_write_delete](https://github.com/postgres/postgres/blob/master/src/backend/replication/logical/proto.c#:~:text=logicalrep_write_delete)

Reply:
> I think existingTuple is a bit confusing in this case, since I would expect that to be the old one.
> 
> Maybe mainTuple or something? (or not touch this switch and do the GetTupleForTargetSchema for each case separately)

Done!